go/types/blob.go

```diff
@@ -30,10 +29,99 @@ func NewEmptyBlob() Blob {
 	return Blob{newBlobLeafSequence(nil, []byte{}), &hash.Hash{}}
 }
 
-// BUG 155 - Should provide Write... Maybe even have Blob implement ReadWriteSeeker
+// ReaderAt interface. Eagerly loads requested byte-range from the blob p-tree.
+func (b Blob) ReadAt(p []byte, off int64) (n int, err error) {
+	// TODO: Support negative off?
+	d.PanicIfTrue(off < 0)
+
+	startIdx := uint64(off)
+	if startIdx >= b.Len() {
+		return 0, io.EOF
+	}
+
+	endIdx := startIdx + uint64(len(p))
+	if endIdx > b.Len() {
+		endIdx = b.Len()
+	}
+
+	if endIdx == b.Len() {
+		err = io.EOF
+	}
+
+	if startIdx == endIdx {
+		return
+	}
+
+	leaves, localStart := LoadLeafSequences(b, startIdx, endIdx)
+	endIdx = localStart + endIdx - startIdx
+	startIdx = localStart
+
+	for _, s := range leaves {
+		bl := s.(blobLeafSequence)
+
+		le := endIdx
```

Comment:
finding this code a bit hard to follow (all the math involving le's and ll's and so forth) but I'm sure it's correct :-)

maybe comments or something would help